lib/streamlit/elements/widgets/button_group.py

```diff
@@ -89,9 +90,29 @@
 SelectionMode: TypeAlias = Literal["single", "multi"]
 
 
-class SingleSelectSerde(Generic[T]):
-    """Uses the MultiSelectSerde under-the-hood, but accepts a single index value
-    and deserializes to a single index value.
+@dataclass
+class _MultiSelectSerde(Generic[T]):
```

Comment:
maybe also add a comment here that this is meant to be used as part of `ButtonGroupSerde` and not on its own.

Reply:
I have added a comment 👍